sql/connect/common/src/main/scala/org/apache/spark/sql/connect/common/config/ConnectCommon.scala

```diff
@@ -16,9 +16,23 @@
  */
 package org.apache.spark.sql.connect.common.config
 
+import io.grpc.Metadata
+
 private[sql] object ConnectCommon {
   val CONNECT_GRPC_BINDING_PORT: Int = 15002
   val CONNECT_GRPC_PORT_MAX_RETRIES: Int = 0
   val CONNECT_GRPC_MAX_MESSAGE_SIZE: Int = 128 * 1024 * 1024
   val CONNECT_GRPC_MARSHALLER_RECURSION_LIMIT: Int = 1024
+  val AUTH_TOKEN_META_DATA_KEY: Metadata.Key[String] =
+    Metadata.Key.of("Authentication", Metadata.ASCII_STRING_MARSHALLER)
+
+  val CONNECT_LOCAL_AUTH_TOKEN_ENV_NAME = "SPARK_CONNECT_LOCAL_AUTH_TOKEN"
+  private var CONNECT_LOCAL_AUTH_TOKEN: Option[String] = Option(
+    System.getenv(CONNECT_LOCAL_AUTH_TOKEN_ENV_NAME))
+  def setLocalAuthToken(token: String): Unit = CONNECT_LOCAL_AUTH_TOKEN = Option(token)
+  def getLocalAuthToken: Option[String] = CONNECT_LOCAL_AUTH_TOKEN
+  def assertLocalAuthToken(token: Option[String]): Unit = token.foreach { t =>
+    assert(CONNECT_LOCAL_AUTH_TOKEN.isDefined)
```

Comment:
Again, you **CANNOT** use asserts for this. They will get elided if you disable assertions. Please throw a proper gRPC exception in the `LocalAuthInterceptor`.

Reply:
For other places, I will fix it later. But here assert is correct because if `token` is set, `CONNECT_LOCAL_AUTH_TOKEN` must be set to for local usage.

Reply:
No, you can't use an assert. Read the comment please. The only way you should hit this code path is through GRPC. You can throw an unauthenticated grpc exception.

Reply:
In Python, you call `SparkSession.stop`, it will only stop `SparkContext` but not the JVM itself in the test. So, later, when you restart `SparkSession`, you can't change the environment variable set to the JVM process.